tests/devices/unit_tests/test_eiger.py

```diff
@@ -282,34 +373,132 @@ def wait_on_staging():
     assert not thread.is_alive()
 
 
-@patch("dodal.devices.eiger.await_value")
+# Tests transition from waiting for stale params to _wait_for_odin_status()
+def test_when_stage_called_then_odin_started_after_stale_params_goes_low(
+    fake_eiger: EigerDetector, mock_set_odin_filewriter, finished_status
+):
+    fake_eiger.odin.file_writer.file_name.set = MagicMock(
+        side_effect=mock_set_odin_filewriter
+    )
+
+    fake_eiger.stale_params.sim_put(1)
+    fake_eiger.odin.file_writer.capture.sim_put(0)
+
+    unwrapped_funcs = [
+        lambda: await_value(fake_eiger.stale_params, 0, 60),
+        fake_eiger._wait_for_odin_status,
+    ]
+
+    wrap_and_do_funcs(unwrapped_funcs)
+
+    assert fake_eiger.odin.file_writer.capture.get() == 0
+
+    fake_eiger.stale_params.sim_put(0)
+
+    await_value(fake_eiger.odin.file_writer.capture, 1).wait(1)
+
+
+# Tests transition from _wait_for_odin_status to cam_acquire_set
+def test_when_stage_called_then_cam_acquired_on_meta_ready(
+    fake_eiger: EigerDetector, mock_set_odin_filewriter, finished_status
+):
+    fake_eiger.odin.file_writer.file_name.set = MagicMock(
+        side_effect=mock_set_odin_filewriter
+    )
+
+    fake_eiger.odin.file_writer.capture.sim_put(0)
+    fake_eiger.stale_params.sim_put(0)
+
+    unwrapped_funcs = [
+        fake_eiger._wait_for_odin_status,
+        lambda: fake_eiger.cam.acquire.set(1, timeout=10),
+    ]
+
+    wrap_and_do_funcs(unwrapped_funcs)
+
+    assert fake_eiger.cam.acquire.get() == 0
+
+    fake_eiger.odin.meta.ready.sim_put(1)
+
+    await_value(fake_eiger.cam.acquire, 1).wait(1)
+
+
+# Tests transition from _wait_fan_ready to _finish_arm
+def test_when_stage_called_then_finish_arm_on_fan_ready(
+    fake_eiger: EigerDetector, mock_set_odin_filewriter, finished_status
+):
+    fake_eiger.odin.file_writer.file_name.set = MagicMock(
+        side_effect=mock_set_odin_filewriter
+    )
+
+    # Mock this function so arming actually finishes
+    fake_eiger.odin.create_finished_status = MagicMock()
+
+    fake_eiger.odin.fan.ready.sim_put(1)
+    fake_eiger.odin.file_writer.capture.sim_put(0)
+
+    unwrapped_funcs = [fake_eiger._wait_fan_ready, fake_eiger._finish_arm]
+    status = wrap_and_do_funcs(unwrapped_funcs)
+
+    fake_eiger.odin.meta.ready.sim_put(1)
+    status.wait(5)
+
+
+@pytest.mark.parametrize(
+    "func",
+    [
+        ("lambda: self.set_detector_threshold(energy=detector_params.current_energy)"),
+        ("fake_eiger.set_cam_pvs"),
+        ("fake_eiger.set_odin_number_of_frame_chunks"),
+        ("fake_eiger.set_odin_pvs"),
+        ("fake_eiger.set_mx_settings_pvs"),
+        ("fake_eiger.set_num_triggers_and_captures"),
+        ("fake_eiger._wait_for_odin_status"),
+        ("lambda: fake_eiger.cam.acquire.set(1, timeout=10)"),
+        ("fake_eiger._wait_fan_ready"),
+        ("fake_eiger._finish_arm"),
+    ],
+)
+def test_check_callback_error(fake_eiger: EigerDetector, func):
```

Comment:
Should: I think a more useful test here would be to keep the same list as defined in the `eiger` but change which one throws an error